packages/compass-home/src/components/home.tsx

```diff
@@ -259,22 +243,6 @@ function Home({
     [appName, connectionInfo]
   );
 
-  const onDataSeviceDisconnected = useCallback(() => {
-    if (!isConnected) {
-      updateTitle(appName);
-      hideCollectionSubMenu();
-      notifyMainProcessOfDisconnect();
-    }
-  }, [appName, isConnected]);
```

Comment:
While it's not the best place for this logic, moving it inside the callback handler for just one certain event listener is changing the logic quite a lot. Instead of doing all those resets every time something causes a disconnect, we now do this only when a menu button is clicked. Is this an expected change to the behavior?

Reply:
So my understanding was that we are disconnecting only from the menu button hence I moved this inside of the callback. I understand that in multi connections realm there is a context menu based way as well but then this would get modified to be happening in a different place. I might be overlooking something and in which case I am more than happy to revert this change.

Reply:
S, as you said, when

Reply:
Good point, I will revert this.